src/peer/negotiator.js

```diff
@@ -606,13 +620,56 @@ class Negotiator extends EventEmitter {
       this._pc.removeStream(localStreams[0]);
     }
 
-    this._replaceStreamCalled = true;
-
-    // a chance to trigger (and do nothing) on removeStream.
-    setTimeout(() => {
+    // HACK: For some reason FF59 doesn't work when Chrome 64 renegotiates after updating the stream.
+    // However, simply updating the localDescription updates the remote stream if the other browser is firefox.
+    // Chrome 64 probably uses replaceTrack-like functions internally.
+    if (this._remoteBrowser === 'firefox') {
       this._pc.addStream(newStream);
-      this._pc.onnegotiationneeded = origOnNegotiationNeeded;
-    });
+
+      // use setTimeout to trigger (and do nothing) on add/removeStream.
+      setTimeout(() => {
+        // update the localDescription with the new stream information (after getting to the right state)
+        let promise;
+        if (this._originator) {
+          promise = this._makeOfferSdp()
+            .then(offer => {
+              return this._pc.setLocalDescription(offer);
+            })
+            .then(() => {
+              return this._pc.setRemoteDescription(this._pc.remoteDescription);
+            })
+            .catch(err => {
+              logger.error(err);
+            });
+        } else {
+          promise = this._pc
+            .setRemoteDescription(this._pc.remoteDescription)
+            .then(() => {
+              return this._pc.createAnswer();
+            })
+            .then(answer => {
+              return this._pc.setLocalDescription(answer);
+            })
+            .catch(err => {
+              logger.error(err);
+            });
+        }
+        // restore onnegotiationneeded in case we need it later.
+        promise.finally(() => {
```

Comment:
Some browsers like Edge and Safari do not have `Proimse#finally` yet.
And this is not the "syntax"(= Babel does not care), so we need some polyfills like [this](https://github.com/zloirock/core-js/blob/v2.5.1/modules/es7.promise.finally.js).

Reply:
This section will only run on chrome 64 so it won't affect edge and safari but I'll just add a then + catch that does the same thing.